stock-portfolio-monitor/src/market/MarketState.ts

```diff
@@ -0,0 +1,269 @@
+import { Instant, LocalDate, LocalTime, ZoneOffset } from "@js-joda/core";
+import { makeAutoObservable } from "mobx";
+import PortfolioState, { Ticker } from "../portfolio/PortfolioState";
+import loadCurrentPrice from "../client/CurrentPriceClient"
+import { loadCandles } from "../client/CandleClient";
+import { sumBy, uniqBy } from "lodash-es";
+
+const MAX_CONSECUTIVE_DAYS_OFF = 5
+
+export default class MarketState {
+  currentPrices: Price[] = []
+  periodInfo: PeriodInfo[] = []
+  period = ViewPeriod.Week
+  graphTicker: Ticker | undefined = "IVV"
+  portfolioState: PortfolioState
+  loading: boolean = false
+  loadingError: string | undefined = undefined
+
+  constructor(portfolioState: PortfolioState) {
+    this.portfolioState = portfolioState
+    makeAutoObservable(this, {
+      portfolioState: false
+    })
+  }
+
+  setPeriod(period: ViewPeriod) {
+    this.period = period
+    this.load()
+  }
+
+  setGraphTicker(ticker: Ticker) {
+    this.graphTicker = ticker
+  }
+
+  *load() {
+    this.loading = true
+    this.loadingError = undefined
+    try {
+      if (this.currentPrices.length === 0) {
+        this.currentPrices = yield Promise.all(this.portfolioState.uniqueTickers.map(async ticker =>
+          ({ ticker: ticker, price: await loadCurrentPrice(ticker) })))
+      }
+      let info = this.periodInfo.find(_ => _.period === this.period)
```

Comment:
Looks like you need `.some` method here instead of `.find`.  And, seems, should be `const` instead of `let`.

Also, underscore (`_`) in arguments is commonly used to show that it's not needed and used. `_ => _.period` looks very weird and confusing.